lib/envars.ml

```diff
@@ -12,7 +12,32 @@ open Util
 
 (** {1 Helper functions} *)
 
-let getenv_else s dft = try Sys.getenv s with Not_found -> dft ()
+(* on Windows env variables are not so common and typically set system wide *)
+let parse_env_line l =
+  try Scanf.sscanf l "%[^=]=%S" (fun name value -> Some(name,value))
+  with _ -> None
+
+let getenv_from_file name =
+  try
+    let ic = open_in "coq_environment.txt" in
+    let rec find () =
+      let l = input_line ic in
+      match parse_env_line l with
+      | Some(n,v) when n = name -> v
+      | _ -> find ()
+    in
+      find ()
+  with
+  | Sys_error s -> raise Not_found
+  | End_of_file -> raise Not_found
+
+let system_getenv name =
+  if Sys.win32 then
+    try Sys.getenv name with Not_found -> getenv_from_file name
```

Comment:
Why only for Windows? The advertised motivation is also valid under other OSes?

Reply:
I was wondering the same thing. @MSoegtropIMC would this be useful also on OSX?

My main concern is that "./coq_environment.txt"  on Unix is a but like looking for troubles, but `/Applications` on OSX is not really Unix...

Reply:
anyway, done